### PR TITLE
Move core to bot, call ex-core code and /end bot command, 2019-09-27 results

### PR DESCRIPTION
**Pull request by KlicOgogo**

### CODE REVIEW

File: bot/main.py [hunk 1]
```diff
@@ -54,13 +62,14 @@ def init(cls):
     def is_active(cls):
         return cls.date
 
-def guard(usernames = None, check_is_active = True):
+
+def guard(usernames=None, check_is_active=True):
     def inner(func):
         def wrapper(message):
-            if not message.chat.id in groups:
+            if message.chat.id not in groups:
```
Comment (Drapegnik) [inline bot/main.py:66]: 👍

File: bot/utils.py [hunk 2]
```diff
@@ -1,11 +1,24 @@
+from collections import Counter as _counter
 import re
 
 # https://regex101.com/r/9dIo49/2
 GAME_REGEX = r'/game @([^\s]+) (\d+):(\d+) @([^\s]+)'
 
+
+def add_result(games, player1, player2, score1, score2, date):
+    new_game = {'Date': date.strftime("%Y-%m-%d"),
+                'Winner': player1 if score1 > score2 else player2,
```
Comment (Drapegnik) [inline bot/utils.py]: what about to replace it with `[(user1, score1), (user2, score2)]` + `sort`?

File: bot/main.py [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
 from settings import token, groups, admins, GAME_FORMAT
-from utils import format_tags, parse_game
+from utils import add_result, format_tags, parse_game
 
 bot = telebot.TeleBot(token)
 logger = telebot.logger
 telebot.logger.setLevel(logging.INFO)
 
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+REPO_ROOT_DIR = os.path.join(SCRIPT_DIR, '..')
```
Comment (Drapegnik) [inline bot/main.py]:
```suggestion
REPO_ROOT_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..')
```

File: bot/utils.py (hunk 2 above)
Comment (Drapegnik) [reply bot/utils.py]: and then:
```py
winner, looser = sortedUsers
```

File: bot/main.py [hunk 4]
```diff
@@ -95,15 +107,23 @@ def game(message):
...
     bot.send_message(message.chat.id, Gameday.getInfo(), parse_mode='markdown')
 
+
 @bot.message_handler(commands=['end'])
 @guard(admins)
 def end(message):
-    bot.reply_to(message, 'TODO: call @klicunou code')
+
+    games_csv_filepath = os.path.join(REPO_ROOT_DIR, 'data/games.csv')
+    games = pd.read_csv(games_csv_filepath)
```
Comment (Drapegnik) [inline bot/main.py:119]: why you need to read `games` here?

Comment (Drapegnik) [reply bot/main.py:119]: could you just [`append`](https://www.guru99.com/reading-and-writing-files-in-python.html#2) new games to file?

**Review by Drapegnik — CHANGES_REQUESTED** [review]

File: bot/utils.py (hunk 2 above)
Comment (KlicOgogo) [reply bot/utils.py]: OK

File: bot/main.py (hunk 4 above)
Comment (KlicOgogo) [reply bot/main.py:119]: I have to calculate a game index, so I need to know previous indexes.

File: bot/main.py [hunk 5]
```diff
@@ -14,8 +14,7 @@
 logger = telebot.logger
 telebot.logger.setLevel(logging.INFO)
 
-SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
-REPO_ROOT_DIR = os.path.join(SCRIPT_DIR, '..')
+REPO_ROOT_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..')
```
Comment (Drapegnik) [inline bot/main.py]: why duplicate? move to `settings.py`

**Review by Drapegnik — CHANGES_REQUESTED** [review]

**Review by Drapegnik — APPROVED** [review]

**Pull request merged**